File: python/7 kyu/302.py
# https://www.codewars.com/kata/5f8a15c06dbd530016be0c19/train/python

# Passed

def duplicate_sandwich(arr):
    seen = set()
    elem_to_start_idx_dict = dict()
    result = []
    for idx, elem in enumerate(arr):
        if elem in seen:
            elem_start_idx = elem_to_start_idx_dict[elem]
            result = arr[elem_start_idx + 1:idx]
            break
            
        elem_to_start_idx_dict[elem] = idx
        seen.add(elem)
    
    return result

output = duplicate_sandwich([0, 1, 2, 3, 4, 5, 6, 1, 7, 8])
print(output)

File: nodejs/7 kyu/302.js
// https://www.codewars.com/kata/585a033e3a36cdc50a00011c/train/javascript

// Passed

function freqSeq(str, sep) {
  let characterToFrequencyMap = new Object();
  for (const char of str) {
    if (!(char in characterToFrequencyMap)) {
      characterToFrequencyMap[char] = 0;
    }
    characterToFrequencyMap[char]++;
  }

  let sequenceArr = str
    .split("")
    .map((char) => String(characterToFrequencyMap[char]));
  let sequence = sequenceArr.join(sep);
  return sequence;
}

const output = freqSeq("hello world", "-");
console.log(output);